### PR TITLE
doc: add the tablet limitation to the manual recovery procedure

### PR DESCRIPTION
**Pull request by annastuchlik**
This PR adds the information that the manual recovery procedure is not supported if tablets are enabled.

In addition, the content in the Manual Recovery Procedure is reorganized by adding the Prerequisites and Procedure subsections - in this way, we can limit the number of Note and Warning boxes that made the page hard to follow.

Fixes https://github.com/scylladb/scylladb/issues/18895

* Backport to 6.0 required. This update is relevant in version 6.0.

### CODE REVIEW

**Review by kbr-scylla — CHANGES_REQUESTED** [review]

File: docs/troubleshooting/handling-node-failures.rst [hunk 1]
```diff
@@ -99,13 +88,29 @@ in the past and then had Raft enabled, and to clusters that were bootstrapped us
...
+**Prerequisites**
+
+* Before proceeding, make sure that the irrecoverable nodes are truly dead, and not, 
+  for example, temporarily partitioned away due to a network failure. If it is 
+  possible for the 'dead' nodes to come back to life, they might communicate and 
+  interfere with the recovery procedure and cause unpredictable problems.
+
+  If you have no means of ensuring that these irrecoverable nodes won't come back 
+  to life and communicate with the rest of the cluster, setup firewall rules or otherwise 
+  isolate your alive nodes to reject any communication attempts from these dead nodes.
```
Comment (bhalevy) [inline docs/troubleshooting/handling-node-failures.rst:103]: @kbr-scylla don't we have the capability to ban nodes that left (or where expelled from) group0?
(out of scope for this PR)

File: docs/troubleshooting/handling-node-failures.rst [hunk 2]
```diff
@@ -99,13 +88,29 @@ in the past and then had Raft enabled, and to clusters that were bootstrapped us
...
 .. note::
 
-   Entering ``RECOVERY`` mode requires a node restart. Restarting an additional node while 
-   some nodes are already dead may lead to unavailability of data queries (assuming that 
-   you haven't lost it already). For example, if you're using the standard RF=3, 
-   CL=QUORUM setup, and you're recovering from a stuck of upgrade procedure because one 
-   of your nodes is dead, restarting another node will cause temporary data query 
-   unavailability (until the node finishes restarting). Prepare your service for 
-   downtime before proceeding.
+  The manual recovery procedure is not supported :doc:`if tablets are enabled on your keyspace </architecture/tablets/>`.
```
[the remark this answers is not in the document]
Comment (kbr-scylla) [reply docs/troubleshooting/handling-node-failures.rst:91]: The more ugly this page is the better, because it is indeed a cursed situation to find oneself in.

**Review by kbr-scylla — APPROVED** [review]
